documentation/golang/README.md

```diff
@@ -677,6 +677,270 @@ published coverage report dashboard such as this one:
          }
       }
 
+
+# Code Coverage with integration tests
+
+## Unit Tests and Coverage
+
+As a continuation of what was explained above, the general idea is that you call "go test" on a particular package and the unit tests in that package are executed. In effect "go test" is a modified compilation pipeline that adds all of the code that runs unit tests (and rather than executing your main.main function it runs the go test code). Unit tests are stored in source files that have names ending with _test.go (which are not compiled normally).
+
+Additionally, you can get this special “test binary” if you pass -c -o <file> to go test, allowing you to run the tests separately (without recompiling the source code each time). Note that the test binary will only include the tests in the package that you explicitly state in the package line (imported packages won’t have their tests run implicitly).
+
+As explained above, when you pass -cover to go test the compiler will add a bunch of instrumentation to your source code to count how often a line of code was executed. So if we want coverage on integration tests we will have to recompile the source code with cover flags. Unfortunately it’s not (easily) possible to get this cover tool to generate such instrumentation using "go build". So if you want code coverage you need to use "go test".
```

Comment:
> when you pass -cover to go test
add a comma (,) after go test
also maybe add quotes around  "go test" in every instance it is used.

Reply:
done